config/checkstyle/tree_walker_module_directives.xml

```diff
@@ -82,6 +82,12 @@
     <property name="format" value="\bString\.format\b"/>
     <property name="message" value="Use string interpolation instead (com.antkorwin:better-strings)"/>
 </module>
+<module name="Regexp">
+    <property name="illegalPattern" value="true"/>
+    <property name="ignoreComments" value="true"/>
+    <property name="format" value="\bupdateUI\b"/>
+    <property name="message" value="Probably unintended updateUI() usage. See docs and consider repaint()."/>
+</module>
 <module name="RegexpSinglelineJava">
```

Comment:
Nit: while you'll here, pls make `RegexpSinglelineJava` vs `Regexp` consistent (unless there's a reason for this distinction)